erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -244,6 +244,23 @@ def set_conversion_rate(self):
 
 	def validate_materials(self):
 		""" Validate raw material entries """
+
+		def get_duplicates(lst):
+			seen = set()
+			seen_add = seen.add
+			for item in lst:
+				if item.item_code in seen or seen_add(item.item_code):
+					yield item
+
+		def get_duplicate_item(lst):
+			try:
+				item = get_duplicates(lst).next()
+				if item:
+					return item
```

Comment:
instead of returning single item return list of duplicate items, e.g. if raw material has items A, B, B, C, C then return list [B, C] as multiple items